src/components/Table.tsx

```diff
@@ -296,9 +286,9 @@ export const Table = <TableRowData extends BaseTableRowData | CustomRowConfig>({
 
           <TableBody
             items={
-              shouldPaginate && list.items.length > pageSize
-                ? list.items.slice(currentPage * pageSize, (currentPage + 1) * pageSize)
-                : list.items
+              shouldPaginate && items.length > pageSize
+                ? items.slice(currentPage * pageSize, (currentPage + 1) * pageSize)
```

Comment:
you could probably also memoize this `items` value since `.slice` will make it a new prop every render